lib/Transforms/Secretize/WrapGeneric.cpp

```diff
@@ -65,27 +68,19 @@ struct WrapWithGeneric : public OpRewritePattern<func::FuncOp> {
             mp.map(opEntryBlock.getArgument(i), blockArguments[i]);
           }
 
-          opEntryBlock.walk([&](Operation *innerOp) {
-            if (!isa<func::ReturnOp>(innerOp)) {
-              b.clone(*innerOp, mp);
-              opsToErase.push_back(innerOp);
-            }
-          });
-
           auto *returnOp = opEntryBlock.getTerminator();
           b.create<secret::YieldOp>(
               loc, llvm::to_vector(llvm::map_range(
                        returnOp->getOperands(),
                        [&](Value v) { return mp.lookupOrDefault(v); })));
+          returnOp->erase();
         });
 
-    rewriter.replaceOp(
-        opEntryBlock.getTerminator(),
-        rewriter.create<func::ReturnOp>(op.getLoc(), newGeneric.getResults()));
-
-    for (auto *erase : llvm::reverse(opsToErase)) {
-      rewriter.eraseOp(erase);
-    }
+    Block &genericBlock = newGeneric.getRegion().front();
+    rewriter.inlineBlockBefore(&opEntryBlock,
```

Comment:
I'm surprised that this doesn't also need an IR mapping. Maybe it's not a problem because in the IR tested, the block being inlined never uses the original function arguments?

I added https://github.com/google/heir/issues/318 to investigate more later.

Reply:
The call for `inlineBlockBefore`'s third argument is the arg values it's using to replace the original block args with, which in this case is the `  genericBlock.getArguments());`

I can add a test case that's smaller but hello world does use the func arguments.

Reply:
Updated the test to include using the func arg.